File: esports-backend/src/main/java/com/esports/controller/AuthController.java
package com.esports.controller;

import com.esports.dto.*;
import com.esports.service.UserService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.NoSuchElementException;

@RestController
@RequestMapping("/api/auth")
@CrossOrigin(origins = "http://localhost:5173")
public class AuthController {

    private final UserService userService;

    public AuthController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping("/register")
    public ResponseEntity<?> register(@Valid @RequestBody RegisterRequest request) {
        AuthResponse response = userService.register(request);
        return ResponseEntity.ok(response);
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@Valid @RequestBody LoginRequest request) {
        AuthResponse response = userService.login(request);
        return ResponseEntity.ok(response);
    }

    @PostMapping("/google")
    public ResponseEntity<?> googleAuth(@Valid @RequestBody GoogleAuthRequest request) {
        try {
            AuthResponse response = userService.googleAuth(request);
            return ResponseEntity.ok(response);
        } catch (IllegalArgumentException e) {
            return ResponseEntity.status(401).body(ApiResponse.error("Google authentication failed"));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Authentication failed"));
        }
    }

    @PostMapping("/update-role")
    public ResponseEntity<?> updateRole(@Valid @RequestBody UpdateRoleRequest request) {
        try {
            AuthResponse response = userService.updateUserRole(request);
            return ResponseEntity.ok(response);
        } catch (Exception e) {
            return ResponseEntity.status(400).body(ApiResponse.error("Failed to update role"));
        }
    }
}

File: esports-backend-dotnet/Upkart1/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using Upkart1.DTO;
using Upkart1.Services;

namespace Upkart1.Controllers
{
    [ApiController]
    [Route("api/auth")]
    public class AuthController : ControllerBase
    {
        private readonly IUserService _userService;

        public AuthController(IUserService userService)
        {
            _userService = userService;
        }

        [HttpPost("register")]
        public IActionResult Register([FromBody] RegisterRequest request)
        {
            var response = _userService.Register(request);
            return Ok(response);
        }

        [HttpPost("login")]
        public IActionResult Login([FromBody] LoginRequest request)
        {
            var response = _userService.Login(request);
            return Ok(response);
        }

        [HttpPost("google")]
        public IActionResult GoogleAuth([FromBody] GoogleAuthRequest request)
        {
            try
            {
                var response = _userService.GoogleAuth(request);
                return Ok(response);
            }
            catch (ArgumentException ex)
            {
                return Unauthorized(new { error = ex.Message });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpPost("update-role")]
        public IActionResult UpdateRole([FromBody] UpdateRoleRequest request)
        {
            try
            {
                var response = _userService.UpdateUserRole(request);
                return Ok(response);
            }
            catch (Exception)
            {
                return BadRequest(ApiResponse<object>.ErrorResponse("Failed to update role"));
            }
        }
    }
}
